tests/func/experiments/test_checkpoints.py

```diff
@@ -188,3 +192,73 @@ def test_resume_non_head_checkpoint(
     )
     new_head = first(results)
     assert orig_branch != dvc.experiments.get_branch_by_rev(new_head)
+
+
+@pytest.mark.parametrize("use_url", [True, False])
+def test_auto_push_during_iterations(
+    tmp_dir, scm, dvc, checkpoint_stage, git_upstream, local_remote, use_url
+):
+    # set up remote repo
+    remote = git_upstream.url if use_url else git_upstream.remote
+    git_upstream.scm.fetch_refspecs(str(tmp_dir), ["master:master"])
+
+    # without auto push
+    results = dvc.experiments.run(checkpoint_stage.addressing)
+    exp = first(results)
+    ref_info = first(exp_refs_by_rev(scm, exp))
+    assert git_upstream.scm.get_ref(str(ref_info)) is None
+
+    # add auto push
+    os.environ["DVC_EXP_AUTO_PUSH"] = remote
+    results = dvc.experiments.run(checkpoint_stage.addressing)
+    assert (tmp_dir / "foo").read_text() == "4"
+    exp = first(results)
+    ref_info = first(exp_refs_by_rev(scm, exp))
+    assert git_upstream.scm.get_ref(str(ref_info)) == exp
+
+    # check the data
+    with git_upstream.dvc.config.edit() as conf:
+        conf["remote"]["local"] = local_remote.config
+        conf["core"]["remote"] = "local"
+
+    git_upstream.dvc.experiments.apply(ref_info.name)
+    git_upstream.dvc.experiments.apply(exp)
+    git_upstream.dvc.pull()
+    assert (git_upstream / "foo").read_text() == "4"
+
+    # resume the remote checkpoint
+    os.environ.pop("DVC_EXP_AUTO_PUSH")
+    with git_upstream.chdir():
+        git_upstream.dvc.experiments.run(checkpoint_stage.addressing)
+    assert (git_upstream / "foo").read_text() == "6"
+
+
+def test_auto_push_error_url(dvc, scm, checkpoint_stage, local_remote):
+    os.environ["DVC_EXP_AUTO_PUSH"] = "none"
+    assert (
+        dvc.experiments.run(checkpoint_stage.addressing, params=["foo=2"])
+        == {}
+    )
+
+
+def test_auto_push_no_remote(dvc, scm, checkpoint_stage, git_upstream):
+    os.environ["DVC_EXP_AUTO_PUSH"] = git_upstream.url
+    assert (
+        dvc.experiments.run(checkpoint_stage.addressing, params=["foo=2"])
+        == {}
+    )
```

Comment:
@dberenbaum 
Currently, all of the remote push errors occur after the stage repro, so the repro will stop after one epoch. I'm not sure what is the best way to handle it.
1. leave the repo to be after epoch1?
2. roll back to the init stage?
3. examine it before stage repro?
4. Don't stop running, just give users a warning.

Reply:
Good point!

From user standpoint, it's better to check before stage repro, right? Is there any downside to that approach for the user?

As far as whether to error or give a warning, considering this feature is primarily for CI, I don't think warnings will be that helpful, so I'd opt for an error. What do you think?